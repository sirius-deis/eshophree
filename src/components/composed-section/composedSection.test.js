import { render, screen } from "@testing-library/react";
import ComposedSection from "./composedSection";

jest.mock('../../hooks/useFetch', () => ({
  __esModule: true,
  default: jest.fn(),
}));

describe("ComposedSection component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should match snapshot", () => {
    const useFetch = require('../../hooks/useFetch').default;
    const mockProducts = [
      { id: 1, name: "Product 1", price: 100 },
      { id: 2, name: "Product 2", price: 200 },
    ];
    useFetch.mockReturnValue([mockProducts, false]);

    const { container } = render(<ComposedSection />);
    expect(container).toMatchSnapshot();
  })

  it("should render loading state initially", () => {
    const useFetch = require('../../hooks/useFetch').default;
    useFetch.mockReturnValue([null, true]);

    render(<ComposedSection />);
    
    expect(screen.queryByText("Top 20")).not.toBeInTheDocument();
    expect(screen.queryByText("Featured Products")).not.toBeInTheDocument();
  });
});