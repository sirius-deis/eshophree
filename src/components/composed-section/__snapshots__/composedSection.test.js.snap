// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComposedSection component should match snapshot 1`] = `
<div>
  <section
    class="sc-crozmw MGLnd"
  >
    <div
      class="sc-jJTsDX kuukQj"
    >
      <div
        class="sc-fFGjHI dwKuma"
      >
        <div
          class="sc-ewnqHT dLOUny"
        >
          <div
            style="display: flex; align-items: center; gap: 0.7rem;"
          >
            <div
              class="sc-gTRrQi eoyTqP"
            >
              Top 20
            </div>
          </div>
          <div>
            <button
              class="sc-gxYJeL sc-kPTPQs kCKvOC"
              disabled=""
            >
              ‹
            </button>
            <button
              class="sc-gxYJeL sc-ispOId kCKvOC"
              disabled=""
            >
              ›
            </button>
          </div>
        </div>
        <div
          class="sc-bgqQcB iPXChy"
          width="100"
        />
        <div
          style="display: flex; gap: 1rem;"
        >
          <div
            style="overflow: hidden;"
          >
            <div
              class="sc-dicizt cTmuUj"
              style="transform: translateX(100%);"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-jJTsDX kuukQj"
    >
      <div
        class="sc-fFGjHI dwKuma"
      >
        <div
          class="sc-ewnqHT dLOUny"
        >
          <div
            style="display: flex; align-items: center; gap: 0.7rem;"
          >
            <div
              class="sc-gTRrQi eoyTqP"
            >
              Featured Products
            </div>
          </div>
          <div>
            <button
              class="sc-gxYJeL sc-kPTPQs kCKvOC"
              disabled=""
            >
              ‹
            </button>
            <button
              class="sc-gxYJeL sc-ispOId kCKvOC"
              disabled=""
            >
              ›
            </button>
          </div>
        </div>
        <div
          class="sc-bgqQcB iPXChy"
          width="100"
        />
        <div
          style="display: flex; gap: 1rem;"
        >
          <div
            style="overflow: hidden;"
          >
            <div
              class="sc-dicizt cTmuUj"
              style="transform: translateX(100%);"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-jJTsDX kuukQj"
    >
      <div
        class="sc-fFGjHI dwKuma"
      >
        <div
          class="sc-ewnqHT dLOUny"
        >
          <div
            style="display: flex; align-items: center; gap: 0.7rem;"
          >
            <div
              class="sc-gTRrQi eoyTqP"
            >
              Top selling Products
            </div>
          </div>
          <div>
            <button
              class="sc-gxYJeL sc-kPTPQs kCKvOC"
              disabled=""
            >
              ‹
            </button>
            <button
              class="sc-gxYJeL sc-ispOId kCKvOC"
              disabled=""
            >
              ›
            </button>
          </div>
        </div>
        <div
          class="sc-bgqQcB iPXChy"
          width="100"
        />
        <div
          style="display: flex; gap: 1rem;"
        >
          <div
            style="overflow: hidden;"
          >
            <div
              class="sc-dicizt cTmuUj"
              style="transform: translateX(100%);"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-jJTsDX kuukQj"
    >
      <div
        class="sc-fFGjHI dwKuma"
      >
        <div
          class="sc-ewnqHT dLOUny"
        >
          <div
            style="display: flex; align-items: center; gap: 0.7rem;"
          >
            <div
              class="sc-gTRrQi eoyTqP"
            >
              On-Sale Products
            </div>
          </div>
          <div>
            <button
              class="sc-gxYJeL sc-kPTPQs kCKvOC"
              disabled=""
            >
              ‹
            </button>
            <button
              class="sc-gxYJeL sc-ispOId kCKvOC"
              disabled=""
            >
              ›
            </button>
          </div>
        </div>
        <div
          class="sc-bgqQcB iPXChy"
          width="100"
        />
        <div
          style="display: flex; gap: 1rem;"
        >
          <div
            style="overflow: hidden;"
          >
            <div
              class="sc-dicizt cTmuUj"
              style="transform: translateX(100%);"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
